utils/misc.ml

```diff
@@ -2023,11 +2027,25 @@ module Maybe_bounded = struct
     | Bounded r when r.bound > 0 -> r.bound <- r.bound - 1
     | Bounded _ -> ()
 
+  let incr = function
+    | Unbounded -> ()
+    | Bounded r -> r.bound <- r.bound + 1
```

Comment:
This should probably have an overflow check, just in case this ever runs on a 32-bit host?